File: Web_Dev/BackEnd/Java/GuiguShang_Bilibili/chapterX_Distributed_System/src/RMI/mixed_sayHello_calculator_GUI/Calculator.java
package RMI.mixed_sayHello_calculator_GUI;

import java.rmi.Remote;
import java.rmi.RemoteException;

/**
 * @author xueshuo
 * @create 2023-04-24 8:00 pm
 */
public interface Calculator extends Remote {
    double add(double a, double b) throws RemoteException;
    double subtract(double a, double b) throws RemoteException;
    double multiply(double a, double b) throws RemoteException;
    double divide(double a, double b) throws RemoteException;
}

File: fullStack/BackEnd/Node/exports-demo/calculator.js
// calculator.js

function add(a, b) {
    return a + b;
}

function subtract(a, b) {
    return a - b;
}

function multiply(a, b) {
    return a * b;
}

function divide(a, b) {
    if (b !== 0) {
        return a / b;
    } else {
        return "Cannot divide by zero!";
    }
}

console.log("hello!")

// Attaching functions to the exports object
exports.add = add;
exports.subtract = subtract;
exports.multiply = multiply;
exports.divide = divide;
